agent/status.go

```diff
@@ -25,15 +25,27 @@ func (agt *Agent) startStatusServer(ctx context.Context, port int) {
 	n := negroni.New()
 	n.Use(negroni.NewRecovery())
 	n.UseHandler(r)
+	srv := &graceful.Server{
+		Timeout: 10 * time.Second,
+		Server: &http.Server{
+			Addr:    addr,
+			Handler: n,
+		},
+	}
+	grip.Infoln("starting status service on:", addr)
 
 	go func() {
-		err := service.RunGracefully(addr, 10*time.Second, n)
-		if err != nil {
-			grip.Notice(err.Error())
-		}
+		srv.ListenAndServe()
 	}()
 
-	grip.Infoln("starting status service on:", addr)
+	go func() {
+		for {
```

Comment:
I don't think you want this `for` or `select`. A minimal example of a `select` with a single `case` deadlocks for me. I also don't think you want to pass the `ctx`, since doesn't that not correctly shut down the server? I think you can just
```
go func() {
        <-ctx.Done()
        srv.Shutdown(context.Background())
}()

Reply:
changed